roles/dns_client/tasks/main.yml

```diff
@@ -0,0 +1,17 @@
+---
+- name: Get OpenShift facts
+  openshift_facts:
+    local_facts:
+      hostname: "{{ openshift_hostname | default(None) }}"
+      ip: "{{ openshift_ip | default(None) }}"
+      public_hostname: "{{ openshift_public_hostname | default(None) }}"
+      public_ip: "{{ openshift_public_ip | default(None) }}"
+
+- name: Configure the resolver
+  template: dest=/etc/resolv.conf src=resolv.conf
```

Comment:
Right. The reason why it's like this is that this was created to build quick test environments: if used, this would be *the* DNS server for the systems in the environment, this is why resolv.conf is a template. Only the internal view is recursive, queries from outside the private network would only resolve the configured authoritative zone. But yes it makes sense to make it more generic.